### PR TITLE
Create cutout fab

### PR DESCRIPTION
**Pull request by bluquar**
This isn't supported by react-native-paper so I "forked" their code and created a custom implementation for the bottom bar cutout.

https://user-images.githubusercontent.com/4309735/150648469-98fbd3b8-900c-4154-9215-244c182f50a7.mov

https://user-images.githubusercontent.com/4309735/150648472-cd54bc17-10ae-432a-a5c5-6af128d89b3e.mov

Also works on iOS and Android.

### CODE REVIEW

File: src/client/create_request/AidRequestCreateDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  const [whatIsNeeded, setWhatIsNeeded] = React.useState<string>('');
+  const areInputsValid = whoIsItFor.length > 0 && whatIsNeeded.length > 0;
+  const [runCreateRequestMutation, createRequestMutationState] = useMutation<
+    CreateAidRequestMutation,
+    CreateAidRequestMutationVariables
+  >(CREATE_AID_REQUEST_MUTATION);
+  const { loading, error } = createRequestMutationState;
+
+  return (
+    <View style={{ height: Dimensions.get('window').height - 50 }}>
```
Comment (lowellbander) [inline src/client/create_request/AidRequestCreateDrawer.tsx:29]: should this `50` be tied to the height of `EndOfListSpacer`?

File: src/client/create_request/AidRequestCreateDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          <Button disabled={loading} mode="text" onPress={closeDrawer}>
+            Cancel
+          </Button>
+        </View>
+      </View>
+      {error != null ? <Text>{error.message}</Text> : null}
+    </View>
+  );
+  async function submit(): Promise<void> {
+    publishToast(undefined);
```
Comment (lowellbander) [inline src/client/create_request/AidRequestCreateDrawer.tsx:63]:
```suggestion
    clearToast();
```

**Review by lowellbander — APPROVED** [review]
so fuckin goooooood

File: src/client/create_request/AidRequestCreateDrawer.tsx (hunk 1 above)
Comment (bluquar) [reply src/client/create_request/AidRequestCreateDrawer.tsx:29]: good idea

File: src/client/create_request/AidRequestCreateDrawer.tsx (hunk 2 above)
Comment (bluquar) [reply src/client/create_request/AidRequestCreateDrawer.tsx:63]: also good idea!

**Pull request merged**